### DISCUSSION TITLE
v2.0.0b2

### DISCUSSION THREAD

**Discussion by mvdwetering** [mvdwetering/huesyncbox#57] [Announcements]
> **Note**
> This is a beta version, hopefully last one before release.
>
> Changes compared to 2.0.0b1
> * Fix brightness entity showing lots of decimals
>
> Changes compared to 2.0.0b0
> * Update/improve README.md
> * Create repair with impacted automations on upgrade from versions below 2.0.0 (so won't trigger if you are already on 2.0.0b0)
> * Change brightness range to 1-100% to match Hue app (was 0-100%)
> * Improve handling of connection issues
> * Fix adding box found by zeroconf flow
> * Update configentry name when name of box changes
> * Add `set_sync_state` service to allow setting of multiple states at once.

# Breaking Changes

> **Warning**
> All automations and dashboards using this integration will need to be updated!
> Make sure you have time for that when upgrading!

* Removed `media_player` entity
* Removed/changed most custom services
* Minimum required Home Assistant Core version is now 2023.8.0

Read the changes section below and/or [README.md](https://github.com/mvdwetering/huesyncbox/blob/rebuild_integration/README.md) for more details.

# Changes

Before this version the functionality of the Philips Hue Play HDMI Sync Box was all exposed through one single mediaplayer entity. Several features of the media_player were abused to control things on the box. E.g. brightness was controlled by the volume slider. While this initially seemed like a good idea (easy controls on the dashboard and only one entity to implement) this turned out to be confusing and quite limiting in the long run. Next to that the way the integration was built did not allow for more entities so it was not possible to extend it. 

The 2.0.0 version is a complete rewrite to allow for multiple entities and modernize the integration in general. Having multiple entities should make it clear what the function of the entity is. And because they are standard Home Assistant entities like `switch`, `number`, `select` and `sensor` they are familiar to most users already.

With the rewrite several features are added that are nowadays standard for integrations, but did not exists back when this integration was created like select entity, reauth, diagnostics, easy enabling of logging and translations for everything. This rewrite also addresses most open issues and feature requests.

Since the integration is now using the latest and greatest Home Assistant features the minimum Home Assistant Core version is now 2023.8.0.

Note that no functionality is lost. It just moved to a different place.

> **Warning** 
> Because of the nature of the changes it means all automations and dashboards using this integration need to be updated.
> A repair will be be created when upgrading that indicates which automations are impacted to help find them.
> Home Assistant will not load automations that use the removed services, so these have to be updated to use the new entities.

To start with, the yaml configuration option has been removed. It has not been used in a long time. If you still have the `huesyncbox:` entry in `configuration.yaml`  it should be removed.

With the rewrite the existing `media_player` entity has been removed and replaced with several new entities.
Below is a table with the new and old situation.

| New entity | Was implemented by |
|---|---|
| Power switch | Media player on/off |
| Light sync switch | Mediaplayer play/pause |
| Dolby Vision compatibility switch | _New feature_ |
| Intensity select | Mediaplayer sound modes |
| Mode select | Mediaplayer next/previous track |
| HDMI input select | Mediaplayer source |
| LED indicator select | _New feature_ |
| Brightness number | Mediaplayer volume or custom `set_brightness` service. Note that the range is now a percentage (1-100%) instead of 0 to 1 |
| HDMI1-4 connection status sensors | `hdmi1-4_status` attributes of mediaplayer |
| Bridge connection status sensor | `bridge_connection_state` attribute of mediaplayer  |
| Bridge ID sensor | `bridge_unique_id` attribute of mediaplayer  |
| IP address | _New feature_ |
| Wifi quality sensor | _New feature_ |

With the transition to the new entities most custom services that were offered became obsolete as native Home Assistant services of the entities can be used now. The table links to the documentation of the standard Home Assistant services that can be used instead.

| Old service name | Replaced by |
|---|---|
| set_brightness | Use services of the brightness [number entity](https://www.home-assistant.io/integrations/number/#services) |
| set_syncmode | Use services of the sync mode [select entity](https://www.home-assistant.io/integrations/select/#services) |
| set_intensity | Use services of the intensity [select entity](https://www.home-assistant.io/integrations/select/#services) |
| set_entertainment_area | Use services of the entertainment area [select entity](https://www.home-assistant.io/integrations/select/#services) |
| set_sync_state | For toggle or cycle options use the services on the relevant entitites. The service can still be used to set multiple states as once for covenience so you don't have to worry about the order in which to set them.  |
| set_bridge | Not changed |

**Full Changelog**: https://github.com/mvdwetering/huesyncbox/compare/v1.23.0...v2.0.0b1

<hr /><em>This discussion was created from the release <a href='https://github.com/mvdwetering/huesyncbox/releases/tag/v2.0.0b2'>v2.0.0b2</a>.</em>

**Comment by poudenes** [comment]
Hi There,

Installed latest Hue Box Integration and when I use the SET option I always get a error , but the SET is working on the Sync Box.

**Version v2.0.0b2**

```
2023-08-20 10:53:38.046 ERROR (MainThread) [homeassistant.helpers.script.websocket_api_script] websocket_api script: Error executing script. Unexpected error for call_service at pos 1: 'ee0a0c4da8e2078287ae04a14dea6f38'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 468, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 703, in _async_call_service_step
    response_data = await self._async_run_long_action(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 665, in _async_run_long_action
    return long_task.result()
           ^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 1974, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2011, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/huesyncbox/services.py", line 93, in async_set_sync_state
    coordinator = hass.data[DOMAIN][config_entry_id]
                  ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'ee0a0c4da8e2078287ae04a14dea6f38'
2023-08-20 10:53:49.379 ERROR (MainThread) [homeassistant.helpers.script.websocket_api_script] websocket_api script: Error executing script. Unexpected error for call_service at pos 1: 'ee0a0c4da8e2078287ae04a14dea6f38'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 468, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 703, in _async_call_service_step
    response_data = await self._async_run_long_action(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 665, in _async_run_long_action
    return long_task.result()
           ^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 1974, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2011, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/huesyncbox/services.py", line 93, in async_set_sync_state
    coordinator = hass.data[DOMAIN][config_entry_id]
                  ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'ee0a0c4da8e2078287ae04a14dea6f38'
```

```
service: huesyncbox.set_sync_state
data:
  sync: true
  brightness: 70
  intensity: moderate
  mode: video
  input: input1
  entertainment_area: TV room
target:
  device_id: ca734212663336331243ffd0104937dc
```

Off-topic: The solution to have this "huesyncbox" set option is great. All in one and everyone can understand this.

**Reply by mvdwetering** [reply]
Could you download the diagnostics from the device page and add the file here (please not copy paste the content, but drag and drop file in the textfield)

![image](https://github.com/mvdwetering/huesyncbox/assets/732514/e31f4926-919a-4d65-971c-0534cd313196)

Could it be that you have somehow multiple config entries linked to the syncbox? E.g. a devicetracker?

**Comment by poudenes** [comment]
Here you are. And where im aware of. Not having device_trackers using at all.

[config_entry-huesyncbox-9aaff23594adc230e0cf931ce460bc48.json.txt](https://github.com/mvdwetering/huesyncbox/files/12388315/config_entry-huesyncbox-9aaff23594adc230e0cf931ce460bc48.json.txt)

**Comment by poudenes** [comment]
Using this:

```
service: huesyncbox.set_sync_state
data:
  sync: true
  brightness: 70
  intensity: moderate
  mode: video
  input: input1
  entertainment_area: TV room
target:
  device_id: ca734212663336331243ffd0104937dc
```

in a automation. Everything that's below this will not be executed because of the errors.

**Reply by mvdwetering** [reply]
Yeah, I think automations stop when an error occurs.

I released 2.0.0b3 which should fix this issue.

**Comment by poudenes** [comment]
Updated to 2.0.0b3. Testing in dev tool gives me no errors. Later the day let me check it with the TV and AppleTV Scenes I have in HA

**Comment by poudenes** [comment]
I can confirm that everything is working correct here.

**Reply by mvdwetering** [reply]
Thanks for confirming. I hope that was the last issue.

**Comment by mvdwetering** [comment]
Thatnks for the feedback during the betas.
v2.0.0 has just been released, please continue disussion there
